topic-operator/src/main/java/io/strimzi/operator/topic/Session.java

```diff
@@ -206,6 +206,8 @@ public void start(Promise<Void> start) {
                     }
 
                 }, "resource-watcher");
+                watcher = new K8sTopicWatcher(topicOperator, initReconcilePromise.future(), resourceThread);
```

Comment:
The thread you pass here is started a couple of line below. This means that the call to `start` in `onClose(KubernetesClientException)` would throw `IllegalThreadStateException`. Which raises several points:

1. We need a test which covers this properly (the existing tests didn't detect the `IllegalThreadStateException`).
2. It's not clear to me now why we need the `resourceThread`. I think the watcher we pass to fabric will be invoked on some thread of fabric8's. Maybe we don't need it at all.
3. If it turned out that we really did need it... I assume `resourceThread` gets stopped when the `Session` is shutdown/closed. That pattern doesn't work if there are other threads which only the `watcher` know about. And I say threads plural because it need to cope with the watcher being closed multiple times, so having a single thread (again, if it's really needed) is not going to work.